docs: migrate 'check a snap's status' (#5342)
canonical/snapcraft: docs/how-to/publish/check-a-snaps-status.rst

Reviewer: Is it worth mentioning that you must be logged in before running these commands?

Author: https://github.com/canonical/snapcraft/pull/5342/commits/ee6ce9f3bd1ad531e74a20e64bf47c2f519cb1f6

    170     2025-03-13T07:59:46Z  amd64     v0.29.0-6-gc772624   latest/edge*
    169     2025-03-11T09:24:42Z  riscv64   v0.29.0-5-g9c64eb4   latest/edge
    168     2025-03-11T08:04:13Z  ppc64el   v0.29.0-5-g9c64eb4   latest/edge
    167     2025-03-11T08:04:12Z  arm64     v0.29.0-5-g9c64eb4   latest/edge
    166     2025-03-11T08:03:13Z  armhf     v0.29.0-5-g9c64eb4   latest/edge
    165     2025-03-11T08:02:12Z  s390x     v0.29.0-5-g9c64eb4   latest/edge
    164     2025-03-11T08:00:13Z  amd64     v0.29.0-5-g9c64eb4   latest/edge
    163     2025-03-02T09:13:12Z  riscv64   v0.28.2-1-geef628d   latest/edge
    162     2025-03-02T08:00:45Z  ppc64el   v0.28.2-1-geef628d   latest/edge
    161     2025-03-02T07:59:38Z  arm64     v0.28.2-1-geef628d   latest/edge
    160     2025-03-02T07:58:30Z  armhf     v0.28.2-1-geef628d   latest/edge
    159     2025-03-02T07:57:11Z  s390x     v0.28.2-1-geef628d   latest/edge
    158     2025-03-02T07:55:11Z  amd64     v0.28.2-1-geef628d   latest/edge
    ...

Like with the channel status, you can filter the revision status for particular
architectures with the ``--arch`` argument:

.. code-block:: bash

>     snapcraft list-revisions <snap-name> --arch amd64

Reviewer 2: Added this passage.

Check a snap's status
=====================

Check the overall snap status

> -----------------------------

Author: Dropped this section because running `snapcraft status`, even without progressive releases in progress, just displays information from the Snap Store. To me, it seemed more appropriate to put that info under "check releases".

If I missed any information that would have fit nicely in this section, please let me know!

Reviewer 2: This was originally meant for `snapcraft names`, but fair point.

Check the status of releases
----------------------------

Run the ``snapcraft status`` command followed by a snap's name to check the status of
its releases.

.. terminal::

>     :input: snapcraft status <snap-name>

Reviewer: I recall hearing guidance to not add prompt characters before a command, since it's common for users to copy/paste the whole line.  Is that true in a context like this with the `terminal` rendering?

Author: I think that is a very important practice in tutorials, as users are expected to copy the commands in exactly as they appear. In this case, I don't think the practice is quite as important, but probably still worth observing. I'll make a change shortly.

Author: https://github.com/canonical/snapcraft/pull/5342/commits/ee6ce9f3bd1ad531e74a20e64bf47c2f519cb1f6

Check a snap's status
=====================

> With Snapcraft, snap authors can easily monitor the status of their published snaps from
> the command-line. This guide highlights commands for checking the status of a snap's
> releases as well as its visibility on the Snap Store.

Reviewer 2: ```suggestion
From the terminal, you can check the status of your published snap, its releases, and
its visibility.

.. If you prefer to monitor your snaps in a web browser, you can `check your snap's
   status on the Snap Store <>`_.
```

The comment anticipates the store how-to.

> Log in to the Snap Store

Reviewer 2: Note to myself: We should probably add this as a how-to, so we don't have to repeat this information in every page about publication.

Author: I was thinking the same thing

------------------------

First, you must log in to the Snap Store with your Ubuntu One account. If you do not
already have an account, create one `here <https://login.ubuntu.com/>`_.

> To log in to an existing account with Snapcraft, run:

Reviewer 2: ```suggestion
To log in to your account in the terminal, run:
```

.. code-block:: yaml

    snapcraft login

> Check the status of releases
> ----------------------------
> 
> To check the status of a snap's releases, run:

Reviewer 2: ```suggestion
Check the status of a snap
--------------------------

A snap's status is comprises of all its releases. When you check the status, the
store returns the state of all channels, tracks, and architectures that you've
published for.

To check the status of a snap, run:
```

Hindsight is 20:<sub>20</sub>.

Reviewer 2: [54c12bd](https://github.com/canonical/snapcraft/pull/5342/commits/54c12bdad2157ba74c500eba1c32642802092691)

.. code-block:: yaml

    snapcraft status <snap-name>

> Assuming you are the owner or a collaborator of the provided snap, ``snapcraft status``
> will display output similar to the following example.

Reviewer 2: ```suggestion
If your account is the author of the selected snap, the output contains
details similar to the following example.
```

.. terminal::
    :input: snapcraft status <snap-name>

    Track    Arch     Channel    Version    Revision
    latest   amd64    stable     61d336a    127
                      candidate  61d336a    127
                      beta       61d336a    127
                      edge       61d336a    127
             arm64    stable     46dbf20    95
                      candidate  46dbf20    95
                      beta       46dbf20    95
                      edge       61d336a    128

> If a progressive release is in progress, ``snapcraft status`` will also display the
> current deployment progress towards the percentages provided when the progressive
> release was created. For example, a progressive release into the stable channel with a
> deployment target of 30% would result in output similar to:

Reviewer 2: ```suggestion
If a progressive release is in progress, the output also contains the current progress
toward its target deployment percentage. For example, a progressive release of the
stable channel with a deployment target of 30% would produce output similar to:
```

.. terminal::
    :input: snapcraft status <snap-name>

    Track     Arch    Channel    Version    Revision    Progress
    latest    amd64   stable     4.1246     341         73 → 70%
                                 4.1249     355         21 → 30%
                      candidate  ↑          ↑           -
                      beta       -          -           -
                      edge       -          -           -

> Check the visibility of the snap
> --------------------------------
> 
> To check the visibility of a snap, run:

Reviewer 2: ```suggestion
Check the public visibility of your snaps
-----------------------------------------

To check the public visibility of all the snaps you registered on the store, run:
```

.. code-block:: yaml

    snapcraft names

> This will display the name, registration date, and visibility status of all snaps

Reviewer 2: ```suggestion
This returns the name, registration date, and visibility of all snaps
```
